lib/FrontendTool/FrontendTool.cpp

```diff
@@ -122,18 +122,26 @@ static bool emitMakeDependenciesIfNeeded(DiagnosticEngine &diags,
   auto escape = [&](StringRef raw) -> StringRef {
     pathBuf.clear();
 
-    static const char badChars[] = " $#:\n";
-    size_t prev = 0;
-    for (auto index = raw.find_first_of(badChars); index != StringRef::npos;
-         index = raw.find_first_of(badChars, index+1)) {
```

Comment:
The purpose of this was to make the loop faster by doing bulk copies of most characters. Is that not worth it?

Reply:
I don't think that it will matter too much in use.  clang (and gcc) both do this.  Plus, if we try to be clever, we need to backtrack in the case that there is a space in the name, and I expect that isn't that unreasonable.  If you think that added complexity is worth it, I can try to revert to that.